rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -423,6 +423,50 @@ def _maybe_create_fee_events(
 
         return events, totals_per_address
 
+    def _handle_self_transfers(
+            self,
+            tx: BitcoinTx,
+            input_totals: dict[BTCAddress, FVal],
+            output_totals: dict[BTCAddress, FVal],
+    ) -> tuple[list[HistoryEvent], dict[BTCAddress, FVal], dict[BTCAddress, FVal]]:
+        """Handle self transfers.
+        Subtracts any self transfer amounts from the input and output totals and creates
+        informational events for each if the address is tracked.
+        Returns a tuple containing the list of events, the adjusted inputs, and the adjusted outputs.
+        """  # noqa: E501
+        events = []
+        # convert items to a list to prevent errors when input_totals is modified during the loop
+        for address, input_amount in list(input_totals.items()):
```

Comment:
instead of doing this, why not use a set to find addresses that appear in input totals & output totals? then continue the flow.